supervisor.go

```diff
@@ -0,0 +1,176 @@
+	return strings.Replace(input, ":supervisorId", supervisorId, 1)
+}
+
+// GetActiveIDs returns array of active supervisor IDs.
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#get-an-array-of-active-supervisor-ids
+func (s *SupervisorService) GetActiveIDs() ([]string, error) {
+	return nil, errors.New("method GetActiveIDs not implemented")
+}
+
+// GetAllActiveSupervisors returns array of active supervisor objects.
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#get-an-array-of-active-supervisor-objects
+func (s *SupervisorService) GetAllActiveSupervisors() ([]SupervisorStateWithSpec, error) {
+	return nil, errors.New("method GetAllActiveSpecs not implemented")
+}
+
+// GetAllActiveStates returns an array of supervisor states.
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#get-an-array-of-supervisor-states
+func (s *SupervisorService) GetActiveStates() ([]SupervisorState, error) {
+	return nil, errors.New("method GetAllActiveStates not implemented")
+}
+
+// GetSpec calls druid Supervisor Status API.
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#get-supervisor-specification
+func (s *SupervisorService) GetSpec(supervisorId string) (OutputIngestionSpec, error) {
+	r, err := s.client.NewRequest("GET", applySupervisorId(supervisorSpecPathPrefix, supervisorId), nil)
+	var result OutputIngestionSpec
+	if err != nil {
+		return result, err
+	}
+	_, err = s.client.Do(r, &result)
+	if err != nil {
+		return result, err
+	}
+	return result, nil
+}
+
+// GetStatus calls druid Supervisor service's Get status API.
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#get-supervisor-status
+func (s *SupervisorService) GetStatus(supervisorId string) (SupervisorStatus, error) {
+	r, err := s.client.NewRequest("GET", applySupervisorId(supervisorStatusPathPrefix, supervisorId), nil)
+	var result SupervisorStatus
+	if err != nil {
+		return result, err
+	}
+	_, err = s.client.Do(r, &result)
+	if err != nil {
+		return result, err
+	}
+	return result, nil
+}
+
+// GetAuditHistory calls druid Supervisor Status API.
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#get-audit-history-for-a-specific-supervisor
+func (s *SupervisorService) GetAuditHistory(_supervisorId string) (interface{}, error) {
+	return nil, errors.New("method GetAuditHistory not implemented")
+}
+
+// GetAuditHistoryAll calls druid Supervisor Status API.
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#get-supervisor-specification
+func (s *SupervisorService) GetAuditHistoryAll() (map[string]OutputIngestionSpec, error) {
+	return nil, errors.New("method GetAuditHistoryAll not implemented")
+}
+
+// Suspend calls druid Supervisor service's Suspend API.
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#suspend-a-running-supervisor
+func (s *SupervisorService) Suspend(_supervisorId string) (OutputIngestionSpec, error) {
```

Comment:
I am a bit confused - is that considered as blank identifier?

Reply:
Initially, yes. I removed it completely in the latest iteration